UnicodeDecodeError: 'ascii' codec can't decode byte 0xe2 in position 15: ordinal not in range(128)

Similar to my [previously reported](https://code.google.com/p/python-nose/issues/detail?id=402) and reportedly fixed issue.

```
$ python --version
Python 2.7.3
$ pip freeze |grep nose
django-nose==1.1
nose==1.2.1
nosexcover==1.0.7
```

If you have an assertion error with unicode in it: 

```
self.assertContains(response, my.name)
```

blows up if `name` is something with unicode like `田中`:

```
Traceback (most recent call last):
File "/Users/poswald/Projects/hats/env/lib/python2.7/site-packages/nose/case.py", line 133, in run
  self.runTest(result)
File "/Users/poswald/Projects/hats/env/lib/python2.7/site-packages/nose/case.py", line 151, in runTest
  test(result)
File "/Users/poswald/Projects/hats/env/lib/python2.7/site-packages/django/test/testcases.py", line 508, in __call__
  super(TransactionTestCase, self).__call__(result)
File "/usr/local/Cellar/python/2.7.3/Frameworks/Python.framework/Versions/2.7/lib/python2.7/unittest/case.py", line 391, in __call__
  return self.run(*args, **kwds)
File "/usr/local/Cellar/python/2.7.3/Frameworks/Python.framework/Versions/2.7/lib/python2.7/unittest/case.py", line 331, in run
  result.addFailure(self, sys.exc_info())
File "/Users/poswald/Projects/hats/env/lib/python2.7/site-packages/nose/proxy.py", line 146, in addFailure
  formatted = plugins.formatFailure(self.test, err)
File "/Users/poswald/Projects/hats/env/lib/python2.7/site-packages/nose/plugins/manager.py", line 99, in __call__
  return self.call(*arg, **kw)
File "/Users/poswald/Projects/hats/env/lib/python2.7/site-packages/nose/plugins/manager.py", line 141, in chain
  result = meth(*arg, **kw)
File "/Users/poswald/Projects/hats/env/lib/python2.7/site-packages/nose/plugins/capture.py", line 86, in formatFailure
  return self.formatError(test, err)
File "/Users/poswald/Projects/hats/env/lib/python2.7/site-packages/nose/plugins/capture.py", line 81, in formatError
  return (ec, self.addCaptureToErr(ev, output), tb)
File "/Users/poswald/Projects/hats/env/lib/python2.7/site-packages/nose/plugins/capture.py", line 92, in addCaptureToErr
  ev = unicode(ev)
UnicodeDecodeError: 'ascii' codec can't decode byte 0xe2 in position 15: ordinal not in range(128)
```

[This call](/nose-devs/nose/blob/master/nose/plugins/capture.py#L92) in `capture.py` blows up:

```
            ev = unicode(ev)
```

because whatever ev is doesn't coerce into unicode properly. For example this blows up:

```
File "/Users/poswald/Projects/hats/env/lib/python2.7/site-packages/nose/plugins/capture.py", line 94, in addCaptureToErr
ev = u'%s' % ev
UnicodeDecodeError: 'ascii' codec can't decode byte 0xe2 in position 15: ordinal not in range(128)
```

But this is fine because the `__repr__` call works and does backslash escaping to return a `str` object:

```
 ev = '%r' % ev
```

outputs: 

```
AssertionError: AssertionError("Couldn't find '\xe2\x98\x82' in response",)
```

Perhaps this is also related to [this python bug](http://bugs.python.org/issue8313)?

Incidentally, your own test cases should probably all include a unicode snowman ☃ or something in them so you can catch this kind of thing.

Let me know if you need more information to debug this.

--- a/functional_tests/support/issue649/test.py
+++ b/functional_tests/support/issue649/test.py
@@ -1,0 +1,9 @@
+# -*- coding: utf-8 -*-
+
+import unittest
+
+class TestUnicodeInAssertion(unittest.TestCase):
+
+    def test_unicodeInAssertion(self):
+        print "Wurst!"
+        raise ValueError("Käse!")

--- a/functional_tests/test_issue_649.py
+++ b/functional_tests/test_issue_649.py
@@ -1,0 +1,18 @@
+# -*- coding: utf-8 -*-
+import os
+import sys
+import unittest
+from nose.plugins.capture import Capture
+from nose.plugins import PluginTester
+
+support = os.path.join(os.path.dirname(__file__), 'support')
+
+class TestIssue649(PluginTester, unittest.TestCase):
+    activate = ''
+    args = ['-v']
+    plugins = [Capture()]
+    suitepath = os.path.join(support, 'issue649')
+
+    def runTest(self):
+        print str(self.output)
+        assert 'UnicodeDecodeError' not in self.output

--- a/nose/plugins/capture.py
+++ b/nose/plugins/capture.py
@@ -89,7 +89,12 @@ class Capture(Plugin):
         if isinstance(ev, BaseException):
             if hasattr(ev, '__unicode__'):
                 # 2.6+
-                ev = unicode(ev)
+                try:
+                    ev = unicode(ev)
+                except UnicodeDecodeError:
+                    # We need a unicode string... take our best shot at getting,
+                    # since we don't know what the original encoding is in.
+                    ev = str(ev).decode('utf8', 'replace')
             else:
                 # 2.5-
                 if not hasattr(ev, 'message'):

